### PR TITLE
Add packages for 2024 UW NASA Hackweek

### PR DESCRIPTION
**Pull request by weiji14**
Updating the conda environment.yml file to include more packages as requested at https://github.com/CryoInTheCloud/hub-image/issues/118.

Newly added:
- hydroeval=0.1.0
- itables=2.14
- jupyter-sshd-proxy=0.2.0
- py-xgboost=2.1.1 (cpu)
~- pytorch=2.3.1 (cpu)~
- stackstac=0.5.1

Extensions that are added back since they have new compatible releases:
- jupyter-offlinenotebook=0.3.1

Upgraded:
- code-server from 4.16.1 to 4.23.1
- snowexsql from 0.4.1 to 0.5.0 (so that we can get geopandas 1.0, xref https://github.com/CryoInTheCloud/hub-image/pull/121#discussion_r1697925958)
- icepyx from 1.1.0 to 1.2.0

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -80,6 +82,9 @@ dependencies:
   - plotly~=5.18.0
   - seaborn~=0.13.2 # statistical plotting with matplotlib
 
+  # Machine Learning packages
```
Comment (weiji14) [inline environment.yml:85]: Consider adding `py-xgboost` (CPU build?) here for https://github.com/geo-smart/website-2024/pull/7.

```suggestion
  # Machine Learning packages
  - py-xgboost~=2.1.1=cpu*
```

**Review by scottyhq — APPROVED** [review]
thanks for the updates @weiji14 !

**Comment by weiji14** [comment]
Will leave this up for another day in case there are any other late additions. @scottyhq, you probably have a better grasp of what's needed across the 3 Hackweek streams, so I'll let you merge this in once you're happy (and ping @tsnow03 to update the hub image afterwards).

**Pull request merged**